server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -277,17 +277,25 @@ public InternalEngine(EngineConfig engineConfig) {
                 );
                 this.localCheckpointTracker = createLocalCheckpointTracker(localCheckpointTrackerSupplier);
                 // TODO: Segrep - should have a separate read only engine rather than all this conditional logic.
-                if (engineConfig.isPrimary()) {
+                if (engineConfig.getIndexSettings().isSegrepEnabled()) {
```

Comment:
Wiring up primary shards is the same in both cases. We can remove some duplication here with:

```
if (engineConfig.getIndexSettings().isSegrepEnabled && engineConfig.isPrimary() == false) {
 ... setup replica
} else {
 ..setup primary
}
```